packages/cli/src/modules/insights/database/repositories/insights-by-period.repository.ts

```diff
@@ -100,19 +109,31 @@ export class InsightsByPeriodRepository extends Repository<InsightsByPeriod> {
 		return aggregationQuery;
 	}
 
+	/**
+	 * Compacts source data into the target period unit
+	 * @param sourceBatchQuery Query builder to get batch source data. Must return those fields: 'id', 'metaId', 'type', 'periodStart', 'value'
+	 * @param sourceTableName The source table name to get source data from
+	 * @param periodUnitToCompactInto the new period unit to compact the data into
+	 */
```

Comment:
I'm fine with this, but I do think it's a better DX to have the docs on the type instead than on the function, because you get the info when you hover over the field, instead of needing to hover over the function:

![image](https://github.com/user-attachments/assets/a2474aae-3eaa-40b1-95b1-0091f6fe8141)

![image](https://github.com/user-attachments/assets/eaa58718-97cb-42aa-8598-1a8ce365f8f6)
vs.
![image](https://github.com/user-attachments/assets/1895f401-c5c0-437d-8aaf-5dd60b6884db)